### DISCUSSION TITLE
supabase.signInWithPassword in Node

### DISCUSSION THREAD

**Discussion by theMK2k** [supabase/supabase#10672] [Questions] [answered]
Hi everyone,

I try to create a CLI application with Node.js which should interact with the supabase instance in the context of a logged-in user. So the application contains:

- SUPABASE_URL
- SUPABASE_ANON_KEY

and asks the user for their email address and password.

However, it seems like the supabase client doesn't work properly in this setup. Even though a signInWithPassword succeeds in getting the proper session, the client itself doesn't run in its context afterwards.

Does anyone of you have an idea if it is somehow possible to "log in" in a node application?

```js
import { createClient } from '@supabase/supabase-js';

const supabase = createClient('<SUPABASE_URL>', '<SUPABASE_ANON_TOKEN>');
const { data, error } = await supabase.auth.signInWithPassword({
  email: '<login email address>',
  password: <login password>',
});

console.log(data); // perfectly fine object with session, access_token etc.

console.log(await supabase.auth.getUser()); // error: AuthApiError: invalid claim: missing sub claim
```

**Comment by silentworks** [comment]
You will need to turn off `persistSession` and probably save the session data that gets returned into a cookie so you can get access to your JWT.

**Reply by jledbet** [reply]
Hi where is "persistSession?"  Also this seemed to be working fine up until a week ago.

**Reply by jledbet** [reply]
Hi - I found this and did createClient(supabaseUrl, supabaseAnonKey, {auth: {persistSession: false}}), but login is still not able to get the user context.

**Comment by kangmingtay** [comment, marked as answer]
Hi @jledbet and @theMK2k, this worked for me: 

```js

import { createClient } from "@supabase/supabase-js";

const supabaseUrl = "https://your-project-ref.supabase.co";
const supabaseKey = "your-anon-key";

const supabase = createClient(supabaseUrl, supabaseKey, {
  auth: {
    persistSession: false,
  },
});
const { data, error } = await supabase.auth.signInWithPassword({
  email: "foo@example.com",
  password: "secret",
});

console.log(data); // perfectly fine object with session, access_token etc.

console.log(await supabase.auth.getUser()); // error: AuthApiError: invalid claim: missing sub claim

```

Also, fwiw, i'm on supabase-js v2.2.0
